src/components/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import {
+    Collapse,
+    Container,
+    Nav,
+    Navbar,
+    NavbarToggler,
+    NavItem
+} from 'reactstrap';
+import { NavLink } from 'react-router-dom';
+import Profile from './Profile';
+
+interface NavBarProps {}
+
+const NavBar: React.FunctionComponent<NavBarProps> = ({}) => {
+    const [isOpen, setIsOpen] = React.useState<boolean>(false);
+
+    const toggle = () => {
+        setIsOpen(!isOpen);
+    };
```

Comment:
Something to consider here would be to bring in `react-use`, which provides a `useToggle` among other nice hook utilities. This is fine, though, if the goal is to keep dependencies minimal.